temporalnexus/operation.go

```diff
@@ -396,3 +457,53 @@ func convertNexusLinks(nexusLinks []nexus.Link, log log.Logger) ([]*common.Link,
 	}
 	return links, nil
 }
+
+func encodeOperationToken(token *nexus.Content) (string, error) {
+	header, err := json.Marshal(token.Header)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal token header: %w", err)
+	}
+	return base64.URLEncoding.EncodeToString(header) + "." + base64.URLEncoding.EncodeToString(token.Data), nil
+}
+
+func decodeOperationToken(encoded string) (*nexus.Content, error) {
+	encodedHeader, encodedData, ok := strings.Cut(encoded, ".")
+	if !ok {
+		return nil, errors.New("token missing period delimiter")
+	}
+	data, err := base64.URLEncoding.DecodeString(encodedData)
+	if err != nil {
+		return nil, err
+	}
+	headerJson, err := base64.URLEncoding.DecodeString(encodedHeader)
+	if err != nil {
+		return nil, err
+	}
+	var header nexus.Header
+	err = json.Unmarshal(headerJson, &header)
+	if err != nil {
+		return nil, err
+	}
+	return &nexus.Content{
+		Header: header,
+		Data:   data,
+	}, nil
+}
+
+func generateWorkflowRunOperationToken(namespace, workflowID string) (string, error) {
+	token := workflowRunOperationToken{
+		NamespaceName: namespace,
+		WorkflowID:    workflowID,
+	}
+	data, err := json.Marshal(token)
```

Comment:
Do we need the mapping from `namespace` + `workflowID` -> token to be deterministic? I am hopping no, but wanted to double check. If the operation token sent from start mismatched with the one in the start request callback header what would happen?

Reply:
Based on our call today we concluded that it's not required to be deterministic.